### PR TITLE
[Machine Learning] Dropout Regularization to handle overfitting in deep learning models.

### LINKED ISSUE #5314: [Machine Learning] Dropout Regularization to handle overfitting in deep learning models.
#### Proposed title of article
[Machine Learning] Dropout Regularization to handle overfitting in deep learning models. 

#### Proposed article introduction
Overfitting and underfitting are common phenomena in the field of machine learning and deep learning. The techniques used to tackle the overfitting problem is called dropout regularization.

Overfitting happens when a model perfectly learns the detail and noise in the training data, this will have a negative impact on the model's performance when it comes to testing data or when the model is exposed to a new environment. The model has a good performance on the training set, but when the model is exposed to a new environment it will give wrong predictions.

Underfitting occurs when the model can neither learn from the training data nor make predictions when exposed to a testing dataset or new environment. This model underperforms both in training and in testing when making predictions.

In this tutorial, we will build a deep learning model using TensorFlow. We will then use the dropout regularization technique to handle underfitting and overfitting. This ensures we have a model that has a good performance on both the training dataset and the testing dataset.
#### Key takeaways 
1. Overfitting vs underfitting.
2. Impacts of overfitting and underfitting model.
3. Building a deep learning model using TensorFlow.
4. Implementing Dropout Regularization.
5. Improving the model performance using Dropout Regularization.

#### Article quality
This is tutorial is unique because we will address both underfitting and overfitting in the deep learning model using a custom model. We will build a custom model using TensorFlow to help explain these problems. We will implement Dropout Regularization so that we can improve the model performance during the training and testing phase. This is a practical approach and a reader can easily follow from scratch. 

#### References
N/A

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by francis966**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5314

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by hectorkambow** [comment]
@srishilesh
Please un-assign this PR. 

You have various in the works - please either publish, close, or push the other along. 
If you need to ping the CM to publish your work - please do.

**Comment by lalith1403** [comment]
plagiarism check done ✔️

**Comment by hectorkambow** [comment]
@section-engineering-education/peer-reviewers 

Is anyone available to review this PR? 

Thank you

**Comment by owinowendy** [comment]
@hectorkambow I can work on this?

**Comment by srishilesh** [comment]
> @srishilesh Please un-assign this PR.
> 
> You have various in the works - please either publish, close, or push the other along. If you need to ping the CM to publish your work - please do.

@hectorkambow 

I can take this up now. I don't have anything else to review currently.

**Review by srishilesh — APPROVED** [review]
@francis966 

Great job! The article was very beginner friendly, the flow seemed to be right, the code snippets were explained properly, and necessary reference links were added.

Your article required very less changes. Please make sure to go through them.

Moving for the final review.

Keep up the great work!

Thanks.

**Comment by ahmadmardeni1** [comment]
Great work team. @francis966 @srishilesh
